File: backend/semgrep_rules/tests/java/kisa_sw_39_null_dereference.java
import java.util.Map;
class kisa_sw_39_null_dereference {
    int unsafe(Map<String,String> map) {
        // ruleid: kisa.sw39.java.nullable-map-result-dereference
        return map.get("name").length();
    }
    int safe(Map<String,String> map) {
        String value = map.get("name");
        // ok: kisa.sw39.java.nullable-map-result-dereference
        return value == null ? 0 : value.length();
    }
}

File: backend/semgrep_rules/tests/python/kisa_sw_39_null_dereference.py
def unsafe(request):
    # ruleid: kisa.sw39.python.nullable-get-result-dereference
    return request.GET.get("name").strip()
def safe(request):
    value = request.GET.get("name")
    # ok: kisa.sw39.python.nullable-get-result-dereference
    return value.strip() if value is not None else ""
